### PR TITLE
Add lock pool metrics

### PR DESCRIPTION
**Pull request by yuzhu**
### What changes are proposed in this pull request?
Add lock pool metrics

### Why are the changes needed?

Give users a way to monitor lock pool usage. 

### Does this PR introduce any user facing changes?
Additional metrics

### CODE REVIEW

**Comment by yuzhu** [comment]
Alleviate https://github.com/Alluxio/alluxio/issues/13959

File: core/common/src/main/java/alluxio/resource/DynamicResourcePool.java [hunk 1]
```diff
@@ -275,6 +276,15 @@ public DynamicResourcePool(Options options) {
...
       }
     }, options.getInitialDelayMs(), options.getGcIntervalMs(), TimeUnit.MILLISECONDS);
+
+    registerGauges();
+  }
+
+  private void registerGauges() {
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricsSystem.getMetricName("ResourcePoolSize_" + getClass().getName() + "_"
+            + Integer.toHexString(System.identityHashCode(this))),
```
Comment (yuzhu) [inline core/common/src/main/java/alluxio/resource/DynamicResourcePool.java:286]: This produces a class type + unique class id string as the name for the gauge.

Comment (jiacheliu3) [reply core/common/src/main/java/alluxio/resource/DynamicResourcePool.java:286]: I wonder how the hex string will help. For example, if you have `ResourcePoolSize_BlockMasterClientPool_abcdef` and `ResourcePoolSize_BlockMasterClientPool_ghijkl`, how do you further associate them back to the resource user? 

nit: `ResourcePoolSize` -> `ResourcePool`?

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Thanks for the improvement! A metric on resource pool is definitely necessary, i'm all in for this idea.

Added some comments on the naming, PTAL

File: core/common/src/main/java/alluxio/resource/DynamicResourcePool.java (hunk 1 above)
Comment (yuzhu) [reply core/common/src/main/java/alluxio/resource/DynamicResourcePool.java:286]: it probably won't, but provides unique naming at the very least.

File: core/common/src/main/java/alluxio/resource/DynamicResourcePool.java [hunk 2]
```diff
@@ -275,6 +276,15 @@ public DynamicResourcePool(Options options) {
         }
       }
     }, options.getInitialDelayMs(), options.getGcIntervalMs(), TimeUnit.MILLISECONDS);
+
+    registerGauges();
+  }
+
+  private void registerGauges() {
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricsSystem.getMetricName("ResourcePoolSize_" + getClass().getName() + "_"
```
Comment (apc999) [inline core/common/src/main/java/alluxio/resource/DynamicResourcePool.java:285]: shall we make this into a util, so other resource pool can follow the same convention to create metrics?

**Comment by yuzhu** [comment]
@jiacheliu3 @apc999 PTAL

**Review by apc999 — APPROVED** [review]

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**